### DISCUSSION TITLE
Keeping images as JPEG (not BMP)

### DISCUSSION THREAD

**Discussion by flommmm** [hluk/CopyQ#2184] [Q&A]
Hey, 

when I copy a JPEG image and paste it, CopyQ delivers JPEG. When I then add e.g. some text into the clipboard, open CopyQ and doubleclick my image again to get it back into clipboard, it will be pasted as BMP and no more as JPEG. Any idea on this?

Best regards
flommmm

**Comment by hluk** [comment]
I think this is due to Qt GUI framework (which is used to access clipboard) converting the image data to other formats.

I haven't tested it but you can try the following automatic command that drops storing BMP if JPEG is available ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Automatic=true
Command="
    copyq:
    const preferredFormat = 'image/jpeg';
    const otherFormat = 'image/bmp';
    if (dataFormats().includes(preferredFormat)) {
        removeData(otherFormat)
    }"
Icon=\xf302
Name=Prefer JPEG to BMP
```
